dcos/mesos.py

```diff
@@ -748,6 +759,12 @@ def __init__(self, task, master):
         self._task = task
         self._master = master
 
+        if 'slave_id' not in self._task or not master:
+            return
+
+        fd = self.fault_domain()
+        self._task.update(fd)
```

Comment:
CI is currently failing, I think you need to add a check for `not None` before this line : 
```
[linux-tests] CMD: ['dcos', 'service', 'log', 'cassandra']
[linux-tests] STDOUT: 
[linux-tests] STDERR: Traceback (most recent call last):
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/subcommand.py", line 101, in run_and_capture
[linux-tests]     exit_code = m.main([self._command] + self._args)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/service/main.py", line 19, in main
[linux-tests]     return _main(argv)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/util.py", line 24, in wrapper
[linux-tests]     result = func(*args, **kwargs)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/service/main.py", line 32, in _main
[linux-tests]     return cmds.execute(_cmds(), args)
[linux-tests]   File "/workspace/dcos-cli/dcos/cmds.py", line 43, in execute
[linux-tests]     return function(*params)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/service/main.py", line 152, in _log
[linux-tests]     return _log_service(follow, lines, service, file_)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/service/main.py", line 178, in _log_service
[linux-tests]     return _log_task(task['id'], follow, lines, file_)
[linux-tests]   File "/workspace/dcos-cli/cli/.tox/py35-integration/lib/python3.5/site-packages/dcoscli/service/main.py", line 204, in _log_task
[linux-tests]     task = mesos.get_master(dcos_client).task(task_id)
[linux-tests]   File "/workspace/dcos-cli/dcos/mesos.py", line 399, in task
[linux-tests]     tasks = self.tasks(fltr, completed)
[linux-tests]   File "/workspace/dcos-cli/dcos/mesos.py", line 471, in tasks
[linux-tests]     task = self._framework_obj(framework).task(task['id'])
[linux-tests]   File "/workspace/dcos-cli/dcos/mesos.py", line 716, in task
[linux-tests]     return self._task_obj(task)
[linux-tests]   File "/workspace/dcos-cli/dcos/mesos.py", line 730, in _task_obj
[linux-tests]     self._tasks[task['id']] = Task(task, self._master)
[linux-tests]   File "/workspace/dcos-cli/dcos/mesos.py", line 766, in __init__
[linux-tests]     self._task.update(fd)
[linux-tests] TypeError: 'NoneType' object is not iterable
```